go/offline/rpc_cache.go

```diff
@@ -44,8 +44,9 @@ func NewRPCCache(g *libkb.GlobalContext) *RPCCache {
 	}
 }
 
-func hash(rpcName string, arg interface{}) ([]byte, error) {
+func hash(rpcName string, uid keybase1.UID, arg interface{}) ([]byte, error) {
 	h := sha256.New()
+	h.Write(uid.ToBytes())
```

Comment:
there's a chance you can collide unless you separate the UID and the rpcName. Maybe it makes sense to:

```go
type hashStruct struct {
    Uid uid
    RpcName string
    Arg interface{}
}
```
and msgpack / hash that, instead.

Reply:
`h.Write(':')` (after line 49, before line 50) would also work, right?

Reply:
how would it collide if uid is fixed length?

Reply:
UID can be empty if nobody is logged in.